test/extensions/filters/http/oauth2/config_test.cc

```diff
@@ -22,33 +22,101 @@ namespace Oauth2 {
+  ON_CALL(secret_manager, findStaticGenericSecretProvider(correctly_loaded_secret))
+      .WillByDefault(Return(std::make_shared<Secret::GenericSecretConfigProviderImpl>(
+          envoy::extensions::transport_sockets::tls::v3::GenericSecret())));
+
+  EXPECT_THROW_WITH_MESSAGE(factory.createFilterFactoryFromProto(*proto_config, "stats", context),
+                            EnvoyException, exception_message);
+}
+
+} // namespace
+
 TEST(ConfigTest, CreateFilter) {
   const std::string yaml = R"EOF(
 config:
-    token_endpoint:
-      cluster: foo
-      uri: oauth.com/token
-      timeout: 3s
-    authorization_endpoint: https://oauth.com/oauth/authorize/
-    redirect_uri: "%REQ(:x-forwarded-proto)%://%REQ(:authority)%/callback"
-    signout_path: 
-      path:
-        exact: /signout
+  token_endpoint:
+    cluster: foo
+    uri: oauth.com/token
+    timeout: 3s
+  credentials:
+    client_id: "secret"
+    token_secret: {}
+    hmac_secret: {}
+  authorization_endpoint: https://oauth.com/oauth/authorize/
+  redirect_uri: "%REQ(:x-forwarded-proto)%://%REQ(:authority)%/callback"
+  redirect_path_matcher:
+    path:
+      exact: /callback
+  signout_path:
+    path:
+      exact: /signout
     )EOF";
 
-  envoy::extensions::filters::http::oauth2::v3alpha::OAuth2 proto_config;
-  MessageUtil::loadFromYaml(yaml, proto_config, ProtobufMessage::getStrictValidationVisitor());
-  NiceMock<Server::Configuration::MockFactoryContext> factory_context;
-  auto& secret_manager = factory_context.cluster_manager_.cluster_manager_factory_.secretManager();
+  OAuth2Config factory;
+  ProtobufTypes::MessagePtr proto_config = factory.createEmptyConfigProto();
+  TestUtility::loadFromYaml(yaml, *proto_config);
+  testing::StrictMock<Server::Configuration::MockFactoryContext> context;
+
+  // This returns non-nullptr for token_secret and hmac_secret.
+  auto& secret_manager = context.cluster_manager_.cluster_manager_factory_.secretManager();
   ON_CALL(secret_manager, findStaticGenericSecretProvider(_))
       .WillByDefault(Return(std::make_shared<Secret::GenericSecretConfigProviderImpl>(
           envoy::extensions::transport_sockets::tls::v3::GenericSecret())));
 
-  OAuth2Config config;
-  auto cb = config.createFilterFactoryFromProtoTyped(proto_config, "whatever", factory_context);
+  EXPECT_CALL(context, messageValidationVisitor());
+  EXPECT_CALL(context, clusterManager());
+  EXPECT_CALL(context, scope());
+  EXPECT_CALL(context, timeSource());
+  EXPECT_CALL(context, api());
+  EXPECT_CALL(context, getTransportSocketFactoryContext());
+  Http::FilterFactoryCb cb = factory.createFilterFactoryFromProto(*proto_config, "stats", context);
+  testing::StrictMock<Http::MockFilterChainFactoryCallbacks> filter_callback;
+  EXPECT_CALL(filter_callback, addStreamDecoderFilter(_));
+  cb(filter_callback);
+}
+
+TEST(ConfigTest, InvalidTokenSecret) {
+  // The token secret is correctly loaded, hence expecting invalid HMAC config.
```

Comment:
Maybe it would be clearer if we specified the failing secret name instead of the passing one?

Reply:
Updated.